test/utility_belt/sql/core_test.clj

```diff
@@ -19,41 +20,122 @@
+  (start-conn!)
+  (next.jdbc/execute! @conn ["drop table people"])
+  (stop-conn!))
+
 (use-fixtures :each (fn [test-fn]
-                      (reset! conn (.start (cp/create connection-spec)))
-                      (setup* @conn)
-                      (test-fn)
-                      (delete-all* @conn)
-                      (.stop @conn)))
+                      (start-conn!)
+                      (try
+                        (setup* @conn)
+                        (delete-all* @conn)
+                        (test-fn)
+                        (finally
+                          (stop-conn!)))))
 
 (deftest crud-operations
   (is (= [] (get-all* @conn)))
-  (is (=  [{:name "yest" :email "test@test.com" :attributes {:bar 1 :foo ["a" "b" "c"]}}]
-          (add* @conn {:name "yest" :email "test@test.com" :attributes {:bar 1 :foo [:a :b :c]}})))
-  (is (= [{:name "who" :email "dat@test.com" :attributes {:bar 1 :foo {:ok "dawg"}}}]
-         (add* @conn {:name "who" :email "dat@test.com" :attributes {:bar 1 :foo {:ok :dawg}}})))
-  (is (= [{:name "who" :email "dat@test.com" :attributes {:bar 1 :foo {:ok "dawg"}}}
-          {:name "yest" :email "test@test.com" :attributes {:bar 1 :foo ["a" "b" "c"]}}]
+  (is (=  [{:name "yest"
+            :email "test@test.com"
+            :attributes {:bar 1
+                         :foo ["a" "b" "c"]}
+            :confirmed_at #inst "2019-06-24"}]
+          (add* @conn {:name "yest"
+
+                       :email "test@test.com"
+
+                       :attributes {:bar 1
+                                    :foo [:a
+                                          :b
+                                          :c]}
+                       :confirmed_at (time/->date-time "2019-06-24")})))
+  (is (= [{:name "who"
+           :email "dat@test.com"
+           :attributes
+           {:bar 1
+            :foo {:ok "dawg"}}
+           :confirmed_at #inst "2018-03-12T00:13:24Z"}]
+         (add* @conn {:name "who"
+                      :email "dat@test.com"
+                      :attributes {:bar 1
+                                   :foo {:ok :dawg}}
+                      :confirmed_at (time/->date-time "2018-03-12T00:13:24Z")})))
+  (is (= [{:name "who"
+           :email "dat@test.com"
+           :attributes {:bar 1
+                        :foo {:ok "dawg"}}
+           :confirmed_at #inst  "2018-03-12T00:13:24Z"}
+          {:name "yest"
+           :email "test@test.com"
+           :attributes {:bar 1
+                        :foo ["a" "b" "c"]}
+           :confirmed_at #inst "2019-06-24"}]
          (get-all* @conn)))
-  (is (= [{:name "yest" :email "test@test.com" :attributes {:bar 1 :foo ["a" "b" "c"]}}]
+  (is (= [{:name "yest"
+           :email "test@test.com"
+           :attributes {:bar 1
+                        :foo ["a" "b" "c"]}
+           :confirmed_at #inst "2019-06-24"}]
          (get-all* @conn {:email "test@test.com"})))
-  (is (= 1
+  (is (= {:next.jdbc/update-count 1}
```

Comment:
this result looks strange

Reply:
That's how it works now :-) And to be honest, I kinda like it - returning just an int always felt somewhat odd. I should be able to bring back previous behavior by configuring the HughSQL+next.jdbc adapter.

That said - I'm figuring out the right settings for `next.jdbc` - atm it's working in a way compatible with `clojure.java.jdbc`, but that makes us write extra layer of code to transform keys coming out of the DB (at minimum we do snake case to kebab case). It is possible to push the key transformations to the driver itself (so it will be faster) - I'll keep digging.

Reply:
Ok, turns out I was using the wrong results directive in the SQL, it most likely affects our existing applications, see here: https://github.com/layerware/hugsql/issues/109